flox-bash/lib/commands/environment.sh

```diff
@@ -1332,14 +1332,18 @@ function floxDestroy() {
 		warn "$i"
 	done
 	if [ $force -gt 0 ] || boolPrompt "Are you sure?" "no"; then
-		# Start by changing to the (default) floxmain branch to ensure
-		# we're not attempting to delete the current branch.
 		if [ -n "$localBranch" ]; then
-			if $invoke_git -C "$environmentMetaDir" checkout --quiet "$defaultBranch" 2>/dev/null; then
-				# Ensure following commands always succeed so that subsequent
-				# invocations can reach the --origin remote removal below.
-				$invoke_git -C "$environmentMetaDir" branch -D "$branchName" || true
-			fi
+
+			# TODO: remove once we move exclusively to the use of bare clones.
+			# Start by changing to the (default) floxmain branch to ensure
+			# we're not attempting to delete the current branch.
+			[ "$($_git -C "$environmentMetaDir" rev-parse --is-bare-repository)" == "true" ] || \
+				$invoke_git -C "$environmentMetaDir" checkout --quiet "$defaultBranch" 2>/dev/null || true
+			# /TODO
```

Comment:
does this need to be a an or?
its a bit confusing to read as `if (...) {} else { invoke git... }`

I think AND would be clearer, but maybe you thought of something i don't

```suggestion
			[ "$($_git -C "$environmentMetaDir" rev-parse --is-bare-repository)" == "false" ] && \
				$invoke_git -C "$environmentMetaDir" checkout --quiet "$defaultBranch" 2>/dev/null || true
			# /TODO
```

Reply:
This is a pattern we use throughout to avoid any command returning a nonzero return code. Using "false" and "&&" would cause this to return nonzero in the default case and then crash the script on account of `set -e`.

Reply:
fair enough, its a bash problem..

thanks for clarifying